specs/networking/p2p-interface.md

```diff
@@ -18,23 +18,88 @@ It consists of four main sections:
 
 - [Network fundamentals](#network-fundamentals)
   - [Transport](#transport)
+      - [Interop](#interop)
+      - [Mainnet](#mainnet)
   - [Encryption and identification](#encryption-and-identification)
-  - [Protocol negotiation](#protocol-negotiation)
+      - [Interop](#interop-1)
+      - [Mainnet](#mainnet-1)
+  - [Protocol Negotiation](#protocol-negotiation)
+      - [Interop](#interop-2)
+      - [Mainnet](#mainnet-2)
   - [Multiplexing](#multiplexing)
 - [Eth2 network interaction domains](#eth2-network-interaction-domains)
   - [Configuration](#configuration)
   - [The gossip domain: gossipsub](#the-gossip-domain-gossipsub)
+    - [Topics and messages](#topics-and-messages)
+      - [Global topics](#global-topics)
+      - [Attestation subnets](#attestation-subnets)
+      - [Interop](#interop-3)
+      - [Mainnet](#mainnet-3)
+    - [Encodings](#encodings)
+      - [Interop](#interop-4)
+      - [Mainnet](#mainnet-4)
   - [The Req/Resp domain](#the-reqresp-domain)
+    - [Protocol identification](#protocol-identification)
+    - [Req/Resp interaction](#reqresp-interaction)
+      - [Requesting side](#requesting-side)
+      - [Responding side](#responding-side)
+    - [Encoding strategies](#encoding-strategies)
+      - [SSZ-encoding strategy (with or without Snappy)](#ssz-encoding-strategy-with-or-without-snappy)
+    - [Messages](#messages)
+      - [Status](#status)
+      - [Goodbye](#goodbye)
+      - [BeaconBlocksByRange](#beaconblocksbyrange)
+      - [BeaconBlocksByRoot](#beaconblocksbyroot)
   - [The discovery domain: discv5](#the-discovery-domain-discv5)
+    - [Integration into libp2p stacks](#integration-into-libp2p-stacks)
+    - [ENR structure](#enr-structure)
+      - [Interop](#interop-5)
+      - [Mainnet](#mainnet-5)
+    - [Topic advertisement](#topic-advertisement)
+      - [Interop](#interop-6)
+      - [Mainnet](#mainnet-6)
 - [Design decision rationale](#design-decision-rationale)
   - [Transport](#transport-1)
+    - [Why are we defining specific transports?](#why-are-we-defining-specific-transports)
```

Comment:
Thoughts on the depth/verbosity here?
Can we tune `doctoc` depth on a per-TOC basis ?

Reply:
Hm, maybe having all of the questions explicitly in TOC is fine..

Reply:
We could make a custom flag for this file, but yeah it's also nice to have verbose TOCs for large files

Reply:
verbose is fine 👍